python/pyarrow/tests/test_dataset.py

```diff
@@ -635,6 +635,37 @@ def test_make_fragment_from_buffer():
     assert pickled.to_table().equals(fragment.to_table())
 
 
+@pytest.mark.parquet
+def test_make_parquet_fragment_from_buffer():
+    import pyarrow.parquet as pq
+
+    table = pa.table([['a', 'b', 'c'],
+                      [12, 11, 10],
+                      ['dog', 'cat', 'rabbit']],
+                     names=['alpha', 'num', 'animal'])
+
+    out = pa.BufferOutputStream()
+    pq.write_table(table, out)
+
+    buffer = out.getvalue()
+
+    formats = [
+        ds.ParquetFileFormat(),
+        ds.ParquetFileFormat(
+            read_options=ds.ParquetReadOptions(
+                use_buffered_stream=True,
+                buffer_size=4096,
```

Comment:
we probably need to use an option that actually alters the output to be able to catch a failure, eg `dictionary_columns`

Reply:
@jorisvandenbossche updated.